com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ModelUtil.java

```diff
@@ -0,0 +1,854 @@
+        }
+    }
+
+    public static int getNumSkins(Scene scene) {
+        return scene.skins.length;
+    }
+
+    public static ArrayList<ModelImporter.Bone> loadSkeleton(Scene scene) {
+        ArrayList<ModelImporter.Bone> skeleton = new ArrayList<>();
+
+        if (scene.skins.length == 0)
+        {
+            return skeleton;
+        }
+
+        // get the first skeleton
+        ModelImporter.Skin skin = scene.skins[0];
+        for (Bone bone : skin.bones) {
+            skeleton.add(bone);
+        }
+
+        return skeleton;
+    }
+
+    public static ArrayList<ModelImporter.Bone> loadSkeleton(byte[] content, String suffix, Options options) {
+        Scene scene = loadScene(content, suffix, options);
+        return loadSkeleton(scene);
+    }
+
+    // Generate skeleton DDF data of bones.
+    // It will extract the position, rotation and scale from the bone transform as needed by the runtime.
+    private static void boneToDDF(ModelImporter.Bone bone, ArrayList<Rig.Bone> ddfBones) {
+        Rig.Bone.Builder b = com.dynamo.rig.proto.Rig.Bone.newBuilder();
+
+        int parentIndex = (bone.parent != null) ? bone.parent.index : -1;
+        b.setParent(parentIndex);
+        b.setId(MurmurHash.hash64(bone.name));
+        b.setName(bone.name);
+
+        b.setLength(0.0f);
+
+        if (bone.node == null) {
+            Transform identityTransform = MathUtil.vecmathIdentityTransform();
+            b.setLocal(identityTransform);
+            b.setWorld(identityTransform);
+        } else {
+            Transform ddfLocal = toDDFTransform(bone.node.local);
+            b.setLocal(ddfLocal);
+
+            Transform ddfWorld = toDDFTransform(bone.node.world);
+            b.setWorld(ddfWorld);
+        }
+
+        Transform ddfBind = toDDFTransform(bone.invBindPose);
+        b.setInverseBindPose(ddfBind);
+
+        ddfBones.add(b.build());
+    }
+
+    public static boolean loadSkeleton(Scene scene, com.dynamo.rig.proto.Rig.Skeleton.Builder skeletonBuilder) {
+        ArrayList<Bone> boneList = loadSkeleton(scene);
+
+        // Generate DDF representation of bones.
+        ArrayList<Rig.Bone> ddfBones = new ArrayList<Rig.Bone>();
+        for (Bone bone : boneList) {
+            boneToDDF(bone, ddfBones);
+        }
+        skeletonBuilder.addAllBones(ddfBones);
+        return ddfBones.size() > 0;
+    }
+
+    public static void skeletonToDDF(ArrayList<ModelImporter.Bone> bones, com.dynamo.rig.proto.Rig.Skeleton.Builder skeletonBuilder) {
+        // Generate DDF representation of bones.
+        ArrayList<Rig.Bone> ddfBones = new ArrayList<>();
+        for (ModelImporter.Bone bone : bones) {
+            boneToDDF(bone, ddfBones);
+        }
+        skeletonBuilder.addAllBones(ddfBones);
+    }
+
+// $ java -cp ~/work/defold/tmp/dynamo_home/share/java/bob-light.jar com.dynamo.bob.pipeline.ModelUtil model_asset.dae
+    public static void main(String[] args) throws IOException {
```

Comment:
For each step on the way, I've tried to make loading/debugging files as easy as possible. See engine/modelc/README.md for more info on how.